### ISSUE TITLE
IndexChecker false warning in length - index situations

### ISSUE THREAD

**Issue by Maxi17** [typetools/checker-framework#2524]
Consider the snippet:
```
import org.checkerframework.checker.index.qual.IndexOrHigh;
import org.checkerframework.checker.index.qual.NonNegative;

public class Testing {
    public @NonNegative int method(int[] array, @IndexOrHigh("#1") int var)  {
        return array.length - var;
    }
}
```
Running javac -processor org.checkerframework.checker.index Testing.java, you will get the following error:
```
Error:(6, 29) java: [return.type.incompatible] incompatible types in return.
  type of expression: @LowerBoundUnknown int
  method return type: @NonNegative int
```

If `var` is a valid index for `array`, or equal to `array.length`, then the value returned is always 0 or greater.
